src/main/scala/lisa/mathematics/SetTheory.scala

```diff
@@ -183,7 +181,7 @@ object SetTheory extends lisa.Main {
         val tay = thenHave((in(z, a), a === y) |- in(z, y)) by RightSubstEq(List((a, y)), lambda(a, in(z, a)))
 
         have((in(z, a), (a === x) \/ (a === y)) |- (in(z, x), in(z, y))) by LeftOr(tax, tay)
-        andThen(applySubst(upairax, false))
+        andThen(Simplify.once(false, upairax))
```

Comment:
This one fails!

Reply:
@SimonGuilloud reminder; this needs to be changed to `Simplify.once(true, upairax)`. It was working before because even given `false`, `applySubst` was running both directions.